src/sagemaker/workflow/pipeline_validation.py

```diff
@@ -0,0 +1,55 @@
+
+from __future__ import absolute_import
+
+from typing import Dict, Union, List
+
+import attr
+
+import botocore.loaders
+from botocore.exceptions import ClientError, ValidationError
+from typing import Any, Dict, List, Sequence, Union, Optional
+
+# from sagemaker.workflow.pipeline import Pipeline
+from sagemaker.workflow.steps import StepTypeEnum
+from sagemaker.workflow.steps import Step
+from sagemaker.workflow.step_collections import StepCollection
+
+class PipelineValidation:
+    """
+
+    Extends pipeline for client-side validation
+
+    Attributes:
+        _shapes_map dict(): dictionary of botocore model restrictions to check at compile time
+
+    """
+
+    _shapes_map = dict()
+
+    def __init__(self,
+                 steps: Sequence[Union[Step, StepCollection]] = attr.ib(factory=list)
```

Comment:
- The `steps` should be a required parameter, so no need to assign the default value.
- We'd better add the `steps` input field into doc string, similar to [here](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/workflow/properties.py#L58-L64)